ir/ir.def

```diff
@@ -178,6 +178,25 @@ class KeyElement {
     Annotations     annotations;
     Expression      expression;
     PathExpression  matchType;
+
+    const IR::Node *transform_visit(Transform &v) {
```

Comment:
I personally prefer to keep the IR classes as small as possible and not to mix visitor and IR code. This makes both the IR and the visitors easier to learn for contributors. Moreover, I expect that there will be exactly 1 pass which needs to do this transformation, so there won't be much reuse for this code.